File: python/leetcode/370.py
class Solution(object):
    def getModifiedArray(self, length, updates):
        res = [0] * length
        for update in updates:
            start, end, inc = update
            res[start] += inc
            if end < length - 1:
                res[end + 1] -= inc
        ans = 0
        for i in range(length):
            ans += res[i]
            res[i] = ans
        return res

File: cpp/leetcode/370.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> getModifiedArray(int length, vector<vector<int>>& updates) {
        vector<int> nums(length, 0);
        for(auto update : updates){
            int start = update[0], end = update[1], inc = update[2];
            nums[start] += inc;
            if(end < length - 1) nums[end + 1] -= inc;
        }
        int ans = 0;
        for(int i = 0; i < length; i++){
            ans += nums[i];
            nums[i] = ans;
        }
        return nums;
    }
};